app/src/main/kotlin/org/kotlinlsp/index/worker/WorkQueue.kt

```diff
@@ -5,6 +5,7 @@ import java.util.concurrent.locks.ReentrantLock
 import kotlin.concurrent.withLock
 
 class WorkQueue<T> {
+    private val scanQueue: Queue<T> = LinkedList()
```

Comment:
I don't see the point of the scan queue, isn't it easier to first do the scanning, and after it's done the background index, so scanning takes less time?